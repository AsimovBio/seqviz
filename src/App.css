#sidebar-container {
  min-height: 100vh;
}

.sidebar-container {
  min-height: 100vh;
}

.control-panel {
  display: flex;
  vertical-align: middle;
  width: 100%;
}

.part-input {
  width: 300px;
  border-right: 1px solid #f0f0f0;
  margin-right: 24px;
}

.part-input input {
  height: 48px;
  box-shadow: none !important;
  border: none !important;
  font-size: 1.2em;
}

.part-input .search.icon {
  margin-right: 8px !important;
}

.part-input button {
  background-color: white !important;
}

.backbone-dropdown {
  margin: auto 24px auto 0;
  vertical-align: middle;
  display: flex;
}

.backbone-dropdown .dropdown {
  margin-right: 8px;
  background: #f0f0f0;
}

.backbone-dropdown .dropdown:hover {
  background: #eaeaea;
}

.backbone-dropdown .dropdown.active {
  background: #e4e4e4 !important;
}

.backbone-dropdown .label {
  margin: auto !important;
}

.backbone-dropdown .dropdown .icon {
  background-color: dodgerblue !important;
  color: aliceblue !important;
}

.selection-meta {
  margin: auto 24px auto 0;
  flex: 1;
}

.selection-meta .label {
  margin-right: 8px !important;
}

.header-logo {
  height: 48px;
  padding-right: 12px;
  margin: auto;
}

.seqviz-container {
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  min-width: 1000px;
}

#header {
  height: 64px;
  padding: 8px 16px;
  margin: auto 0;
  box-shadow: 0 2px 8px #f0f1f2;
  min-width: 1000px;
}

#sidebar-header {
  height: 64px;
  padding: 8px 16px;
  margin: auto 0;
  box-shadow: 0 2px 8px #f0f1f2;
}

#seqviewer {
  /* height is 100% of the viewport height less the header height */
  height: calc(100% - 64px);
  width: 100%;
  margin: 0px;
  flex: 1 1 0;
}

#selection-name {
  font-weight: bold;
  font-style: italic;
}

#selection-type {
  font-style: italic;
}

#selection-label {
  font-weight: bold;
}

#selection-name,
#selection-type,
#selection-length,
#selection-range,
#sequence-gc,
#sequence-tm {
  padding: 5px;
}

#selection-label {
  padding: 5px 0px 5px 5px;
}
