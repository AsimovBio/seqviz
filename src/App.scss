#part-explorer {
  height: 100%;
  .seq-viewers-container {
    display: flex;
    height: inherit;
  }
}

label {
  width: 100%;
  font-weight: $regular;
}

.select {
  background-color: transparent;
  outline: none;
  font-size: 13px;
  height: 30px;
  width: 168px;
  padding-left: 5px;
  border: 1px solid #bebebe;
  padding: 5px 10px;
  color: black;
  float: right;
}

.input {
  width: auto;
  padding-left: 5px;
  color: black;
  background-color: $inputbkg;
  border: none;
  padding: 5px 10px;
  border-radius: 1px;
  margin-left: auto;
  outline: none;
}

.modal {
  box-shadow: 2px 2px 12px rgba(0, 0, 0, 0.2);
  border-radius: 3px;
  background-color: white;
}

.scroll::-webkit-scrollbar {
  width: 7px;
}

.scroll::-webkit-scrollbar-thumb {
  background-clip: padding-box;
  border-radius: 7px;
  background-color: $scrollthumbsidebar;
}

.scroll::-webkit-scrollbar-track {
  border-radius: 20px;
}

.scroll::-webkit-scrollbar-track-piece {
  background-clip: padding-box;
  border-radius: 7px;
  background-color: $scrolltracksidebar;
}
