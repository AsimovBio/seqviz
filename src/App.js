import ReactDOM from "react-dom";
import ReactDOMServer from "react-dom/server";
import React from "react";
import PartExplorer from "./PartExplorer/PartExplorer.jsx";
import "./App.scss";

const Viewer = (element, part, options) => {
  const displayConfiguration = false;
  const displayConfig = (displayPart, displayOptions) => {
    const {
      annotate,
      viewer: viewerType,
      showAnnotations,
      showComplement,
      showIndex,
      colors,
      zoom,
      backbone,
      searchQuery: { query, mismatch },
      enzymes
    } = displayOptions;

    const displayName = displayPart.name
      ? displayPart.name
      : displayPart.constructor.name === "FileList"
      ? displayPart[0].name
      : displayPart;
    const displayType = viewerType;
    const displayAnnotate = annotate ? "on" : "off";
    const displayAnnotations = showAnnotations ? "on" : "off";
    const displayComplement = showComplement ? "on" : "off";
    const displayIndex = showIndex ? "on" : "off";
    const displayCustomColors = colors.length ? "yes" : "no";
    const displayZoomLinear =
      zoom.linear > 50
        ? zoom.linear - 50
        : zoom.linear < 50
        ? 0 - (50 - zoom.linear)
        : 0;
    const displayBackbone =
      displayName.startsWith("BB") && backbone.length
        ? `BioBrick Backbone : ${backbone}`
        : "";
    console.log(
      `
    ====================================================
    Current Part: ${displayName}
    Current SeqViz Settings:
        Viewer Type: ${displayType} (circular | linear | both)
        Auto-annotation: ${displayAnnotate}
        Show Annotations: ${displayAnnotations}
        Show Complement: ${displayComplement}
        Show Index: ${displayIndex}
        Using Custom Colors: ${displayCustomColors}
        Linear Zoom: ${displayZoomLinear} (-50 . 50)
        Searching for sequence "${query}" with ${mismatch} mismatch allowance
        Showing cut sites for enzymes: ${enzymes}
        ${displayBackbone}
    =====================================================
    `
    );
    if (viewerType === "circular" && query !== "") {
      console.warn(
        "Search visualization is only supported in Linear Sequence View."
      );
    }
  };

  const viewer = <PartExplorer part={part} {...options} />;
  const viewerHTML = ReactDOMServer.renderToString(viewer);

  const domElement =
    element.constructor.name.startsWith("HTML") &&
    element.constructor.name.endsWith("Element")
      ? element
      : document.getElementById(element);

  const render = () => {
    ReactDOM.render(viewer, domElement);
  };

  if (displayConfiguration) displayConfig(part, options);
  return {
    viewer: viewer,
    viewerHTML: viewerHTML,
    render: render
  };
};

export default Viewer;
