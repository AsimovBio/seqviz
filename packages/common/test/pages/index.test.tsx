import React from 'react';
import App from '../../pages/index';
import { render } from '../testUtils';

describe('App', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<App />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
