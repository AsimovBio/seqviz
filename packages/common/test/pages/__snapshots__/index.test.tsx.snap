// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main />
    <footer />
  </div>
</DocumentFragment>
`;
